openedx/core/djangoapps/credit/partition_schemes.py

```diff
@@ -53,18 +54,47 @@ def get_group_for_user(cls, course_key, user, user_partition):
             string of allowed access group
         """
         checkpoint = user_partition.parameters['location']
+        # here getting cache key names for all models. So that we can make the
+        # list of keys and get the cache.get_many
 
-        if (
-                not is_enrolled_in_verified_mode(user, course_key)
-        ):
+        enrollment_cache_key = CourseEnrollment.cache_key_name(user.id, unicode(course_key))
+        has_skipped_cache_key = SkippedReverification.cache_key_name(user.id, unicode(course_key))
+        verification_status_cache_key = VerificationStatus.cache_key_name(user.id, unicode(course_key))
+
+        cache_keys = [
+            enrollment_cache_key, has_skipped_cache_key, verification_status_cache_key
+        ]
+
+        cache_values = cache.get_many(cache_keys)
+
+        if enrollment_cache_key not in cache_values:
+            is_verified = is_enrolled_in_verified_mode(user, course_key)
+        else:
+            enrollment_mode = cache_values[enrollment_cache_key]
+            is_verified = enrollment_mode in CourseMode.VERIFIED_MODES
+
+        if has_skipped_cache_key not in cache_values:
+            has_skipped = has_skipped_any_checkpoint(user, course_key)
+        else:
+            has_skipped = cache_values[has_skipped_cache_key]
+
+        if verification_status_cache_key not in cache_values:
+            verification_statuses = VerificationStatus.get_all_checkpoints(user.id, course_key)
+        else:
+            verification_statuses = cache_values[verification_status_cache_key]
```

Comment:
This approach does not work. As it always execute the default function. So either cache value exists or not there is call for db.